src/igr-tests/before-after/run-test.sh

```diff
@@ -1,46 +1,44 @@
 #!/bin/sh
 
 set -eu
-
-DINIT_EXEC=${DINIT_EXEC:-../../dinit}
-DINITCTL_EXEC=${DINITCTL_EXE:-../../dinitctl}
-
-cd "$(dirname "$0")"
+. "$IGR_FUNCTIONS"
 
 # Tests around before/after link functionality.
 
-mkdir -p output
-rm -rf output/*
+rm -rf "$IGR_OUTPUT"/output/*
 
-"$DINIT_EXEC" -q -d sd1 -u -p socket  &
+spawn_dinit -d sd1
 
-# Give some time for startup
-sleep 0.2
+# Custom error.
+__error() {
+    stop_dinit
+    exit 1
+}
 
 # start parent; should start service2 and then service1 (due to before= in service2).
-"$DINITCTL_EXEC" --quiet -p socket start parent
+dinitctl $QUIET start parent || __error
 
-if [ "$(cat output/script-output)" != "$(printf "two\none\n")" ]; then
-    "$DINITCTL_EXEC" --quiet -p socket shutdown
-    return 1
+if ! compare_text "$IGR_OUTPUT"/output/script-output "$(printf "two\none\n")"; then
+    stop_dinit
+    error "$IGR_OUTPUT/output/script-output didn't contain expected result!"
```

Comment:
`error` already calls `stop_dinit` so why do both? It looks like if `stop-dinit` is called twice it will cause an error